src/edu/sjsu/stratagem/ExpressionBuilderVisitor.java

```diff
@@ -106,39 +75,79 @@ public Expression visitFunctionApp(StratagemParser.FunctionAppContext ctx) {
 
 	@Override
 	public Expression visitFunctionDecl(StratagemParser.FunctionDeclContext ctx) {
-		List<String> params = new ArrayList<>();
-		for (int i=1; i<ctx.params().getChildCount()-1; i+=2) {
-			params.add(ctx.params().getChild(i).getText());
-		}
+		List<String> paramNames = new ArrayList<>();
+		paramNames.add(ctx.params().ID().getText());
+
+		List<Type> paramTypes = new ArrayList<>();
+		paramTypes.add(parseType(ctx.params().type()));
+
+		Type returnType = parseType(ctx.type());
 		Expression body = visit(ctx.seq());
-		return new FunctionDeclExpr(params, body);
+
+		return new FunctionDeclExpr(paramNames, paramTypes, returnType, body);
+	}
+
+	@Override
+	public Expression visitId(StratagemParser.IdContext ctx) {
+		String id = ctx.ID().getText();
+		return new VarExpr(id);
+	}
+
+	@Override
+	public Expression visitIf(StratagemParser.IfContext ctx) {
+		Expression cond = visit(ctx.expr());
+		Expression thn = visit(ctx.seq(0));
+		Expression els = visit(ctx.seq(1));
+		return new IfExpr(cond, thn, els);
+	}
+
+	@Override
+	public Expression visitInt(StratagemParser.IntContext ctx) {
+		int val = Integer.valueOf(ctx.LIT_INT().getText());
+		return new ValueExpr(new IntVal(val));
 	}
 
 	@Override
 	public Expression visitLet(StratagemParser.LetContext ctx) {
 		String id = ctx.ID().getText();
 		Expression value = visit(ctx.expr(0));
-		Expression next = visit(ctx.expr(1));
+		Expression body = visit(ctx.expr(1));
+
+		List<String> paramNames = new ArrayList<>();
+		List<Type> paramTypes = new ArrayList<>();
+
+		paramNames.add(id);
+		paramTypes.add(parseType(ctx.type()));
 
-		List<String> params = new ArrayList<>();
-		params.add(id);
-		FunctionDeclExpr implicitDecl = new FunctionDeclExpr(params, next);
+		// FIXME: Need to determine return type somehow. Add an explicit annotation in the grammar?
+		Type returnType = null;
+
+		FunctionDeclExpr implicitDecl = new FunctionDeclExpr(paramNames, paramTypes, returnType, body);
 
 		List<Expression> args = new ArrayList<>();
 		args.add(value);
 		return new FunctionAppExpr(implicitDecl, args);
 	}
 
 	@Override
-	public Expression visitInt(StratagemParser.IntContext ctx) {
-		int val = Integer.valueOf(ctx.LIT_INT().getText());
-		return new ValueExpr(new IntVal(val));
+	public Expression visitPrint(StratagemParser.PrintContext ctx) {
+		Expression arg = visit(ctx.args().getChild(1));
+	    return new PrintExpr(arg);
```

Comment:
This indentation should be fixed. There's an issue with my editor. Also, some files are indented with space and other tabs... I might unify them on one standard.

Do you have any preference for which?